tests/reserve_tests.cpp

```diff
@@ -0,0 +1,192 @@
+         ("derivative", "10000.00 GXCP@gamex")
+         ("underlying", "100000.0000 HOBL")
+         ("rate", "10.00000000000000000")
+   );
+
+   BOOST_REQUIRE_EQUAL(wasm_assert_msg("additional issuance not supported yet"),
+         mint(EA("1000.00 GXCP@gamex"), EA("100000.0000 HOBL@gxc"))
+   );
+
+} FC_LOG_AND_RETHROW()
+
+BOOST_FIXTURE_TEST_CASE(prepay_tests, gxc_reserve_tester) try {
+
+   gxc_token_tester::mint(EA("10000.0000 GXC@gxc"), false);
+   setminamount(EA("10.0000 GXC@gxc"));
+   transfer(config::null_account_name, N(gxc), EA("100.0000 GXC@gxc"), "hola");
+   approve(N(gxc), reserve_account_name, EA("100.0000 GXC@gxc"));
+   produce_blocks(1);
+
+   BOOST_REQUIRE_EQUAL(wasm_assert_msg("not enough to prepay"),
+         prepay(N(gxc), N(gamex), EA("1.0000 GXC@gxc"))
+   );
+
+   BOOST_REQUIRE_EQUAL(success(),
+         prepay(N(gxc), N(gamex), EA("100.0000 GXC@gxc"))
+   );
+   produce_blocks(1);
+
+   REQUIRE_MATCHING_OBJECT(get_table_row(reserve_account_name, reserve_account_name, N(prepaid), N(gamex)), mvo()
+         ("creator", "gxc")
+         ("name", "gamex")
+         ("value", "100.0000 GXC@gxc")
+   );
+   produce_blocks(1);
+
+} FC_LOG_AND_RETHROW()
+
+BOOST_FIXTURE_TEST_CASE(reserve_claim_tests, gxc_reserve_tester) try {
+
+   set_authority(token_account_name, "active",
+         authority(1,
+            vector<key_weight>{{get_private_key("gxc.token", "active").get_public_key(), 1}},
+            vector<permission_level_weight>{{{N(gxc), config::active_name}, 1}, {{N(gxc.reserve), config::eosio_code_name}, 1}, {{N(gxc.token), config::eosio_code_name}, 1}}, {}),
+         config::owner_name,
+         {{token_account_name, config::active_name}},
+         {get_private_key(token_account_name, "active")}
+   );
+
+   set_authority(reserve_account_name, "active",
+         authority(1,
+            vector<key_weight>{{get_private_key("gxc.reserve", "active").get_public_key(), 1}},
+            vector<permission_level_weight>{{{N(gxc), config::active_name}, 1}, {{N(gxc.reserve), config::eosio_code_name}, 1}}, {}),
+         config::owner_name,
+         {{reserve_account_name, config::active_name}},
+         {get_private_key(reserve_account_name, "active")}
+   );
+   produce_blocks(1);
+
+   gxc_token_tester::mint(EA("1000000.0000 HOBL@gxc"));
+   setminamount(EA("10.0000 HOBL@gxc"));
+   produce_blocks(1);
+
+   setpartner(N(gamex));
+   transfer(config::null_account_name, N(gamex), EA("100000.0000 HOBL@gxc"), "hola");
+   approve(N(gamex), N(gxc.reserve), EA("100000.0000 HOBL@gxc"));
+   mint(EA("10000.00 GXCP@gamex"), EA("100000.0000 HOBL@gxc"));
+   produce_blocks(1);
+
+   transfer(config::null_account_name, N(gamex), EA("1000.00 GXCP@gamex"), "hola", N(gamex));
+   approve(N(gamex), N(gxc.reserve), EA("100.00 GXCP@gamex"));
+   produce_blocks(1);
+
+   claim(N(gamex), EA("100.00 GXCP@gamex"));
```

Comment:
After claim, burnt token status and claimed token balance also need to be checked.